File: test_frontend.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();
  
  page.on('console', msg => console.log('BROWSER LOG:', msg.text()));
  page.on('pageerror', err => console.log('BROWSER ERROR:', err));

  await page.goto('http://127.0.0.1:5000');
  
  console.log('Clicking scrape button...');
  await page.click('#btnScrape');
  
  // Wait for either the table to appear, or an alert
  page.on('dialog', async dialog => {
    console.log('ALERT TRIGGERED:', dialog.message());
    await dialog.accept();
  });

  console.log('Waiting for liveStatus to finish...');
  try {
    // Wait until liveStatus display becomes none OR tableWrapper becomes block
    await page.waitForFunction(() => {
      const live = document.getElementById('liveStatus').style.display;
      const table = document.getElementById('tableWrapper').style.display;
      return live === 'none' || table === 'block';
    }, { timeout: 60000 });
  } catch (e) {
    console.log('Wait timeout or error:', e);
  }

  const tableDisplay = await page.evaluate(() => document.getElementById('tableWrapper').style.display);
  const liveStatus = await page.evaluate(() => document.getElementById('liveStatus').style.display);
  const liveStatusText = await page.evaluate(() => document.getElementById('liveStatusText').textContent);
  
  console.log('Table display:', tableDisplay);
  console.log('LiveStatus display:', liveStatus);
  console.log('LiveStatus Text:', liveStatusText);
  
  const html = await page.evaluate(() => document.getElementById('tableBody').innerHTML);
  console.log('Table HTML:', html.substring(0, 200));

  await browser.close();
})();

File: test_frontend.py
from playwright.sync_api import sync_playwright

def run_test():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        
        page.on("console", lambda msg: print(f"BROWSER LOG: {msg.text}"))
        
        print("Navigating to http://127.0.0.1:5000...")
        page.goto("http://127.0.0.1:5000")
        
        page.on("dialog", lambda dialog: print(f"ALERT TRIGGERED: {dialog.message}") or dialog.accept())
        
        print("Clicking scrape button...")
        page.locator("#btnScrape").click()
        
        print("Waiting for scrape to finish...")
        try:
            page.wait_for_function('''() => {
                const live = document.getElementById('liveStatus').style.display;
                const table = document.getElementById('tableWrapper').style.display;
                return live === 'none' || table === 'block';
            }''', timeout=60000)
        except Exception as e:
            print(f"Wait timeout or error: {e}")
            
        table_display = page.evaluate("document.getElementById('tableWrapper').style.display")
        live_status = page.evaluate("document.getElementById('liveStatus').style.display")
        live_text = page.evaluate("document.getElementById('liveStatusText').textContent")
        
        print(f"Table display: {table_display}")
        print(f"LiveStatus display: {live_status}")
        print(f"LiveStatus Text: {live_text}")
        
        html = page.evaluate("document.getElementById('tableBody').innerHTML")
        print(f"Table HTML: {html[:200]}")
        
        browser.close()

if __name__ == "__main__":
    run_test()
